test/message_test.go

```diff
@@ -2351,4 +2352,316 @@ var _ = Describe("Remote Message Tests", func() {
+					// check the published message via semp
+					client := helpers.GetClient(messagingService)
+					Expect(client.DataTxMsgCount).To(Equal(int64(1)))                                  // should be only one message published
+					Expect(client.DataRxMsgCount).To(Equal(int64(1)))                                  // should be only one message received
+					Expect(publishMessageBytesSize).To(BeNumerically("==", len(receivedMessageBytes))) // received should be equal which is zero
+					Expect(client.DataTxByteCount).To(BeNumerically("==", client.DataRxByteCount))     // Tx and Rx data on wire should be the same (no message corruption)
+					Expect(client.DataTxByteCount).To(BeNumerically(">", len(receivedMessageBytes)))   // Tx data on wire should be larger than message without payload (only headers)
+
+				case <-time.After(1 * time.Second):
+					Fail("timed out waiting for message to be delivered")
+				}
+			})
+
+			// Test that string payload is compressed when level is set (set to 7)
+			// String compression performs worse when the letter frequency of repeating characters is very small
+			It("payload should be properly compressed/decompressed after publish/receive with a string payload", func() {
+				payload := strings.Repeat("hello world", 200)
+				message, err := messageBuilder.BuildWithStringPayload(payload)
+				Expect(err).ToNot(HaveOccurred())
+
+				publishMessagePayload, _ := message.GetPayloadAsString()
+				publishMessagePayloadSize := len(publishMessagePayload)
+				publisher.Publish(message, resource.TopicOf(topic))
+
+				select {
+				case message := <-inboundMessageChannel:
+					content, ok := message.GetPayloadAsString()
+					Expect(ok).To(BeTrue())
+					Expect(content).ToNot(BeNil())
+					msgBytes, _ := message.GetPayloadAsBytes()
+					Expect(msgBytes).ToNot(BeNil())
+					// check the published message via semp
+					client := helpers.GetClient(messagingService)
+					Expect(client.DataTxMsgCount).To(Equal(int64(1)))                              // should be only one message published
+					Expect(client.DataRxMsgCount).To(Equal(int64(1)))                              // should be only one message received
+					Expect(publishMessagePayloadSize).To(BeNumerically("==", len(content)))        // bytes received should be equal to published
+					Expect(client.DataTxByteCount).To(BeNumerically("==", client.DataRxByteCount)) // Tx and Rx data on wire should be the same (no message corruption)
+					Expect(client.DataTxByteCount).To(BeNumerically("<", len(msgBytes)))           // Tx data on wire should be Smaller than message when payload is compressed
+
+				case <-time.After(1 * time.Second):
+					Fail("timed out waiting for message to be delivered")
+				}
+			})
+
+			// Test that byte array payload is compressed when level is set (set to 7)
+			It("payload should be properly compressed/decompressed after publish/receive with a byte array payload", func() {
+				largeByteArray := make([]byte, 16384)
```

Comment:
To be clear here the goal is to create a large message payload where all values are 0? I believe make([]byte, 16384) does this. If not there is a chance that random heap memory might lead to a growth in payload.